### PR TITLE
feat(html): Add escape and unescape functions for HTML entities

### PR DESCRIPTION
**Pull request by lionel-rowe**
Fixes https://github.com/denoland/deno_std/issues/3329

### CODE REVIEW

**Comment by lino-levan** [comment]
Maybe we should rename `html_entities` to just `entities` since it is in the html top level folder?

**Comment by iuioiua** [comment]
> Maybe we should rename `html_entities` to just `entities` since it is in the html top level folder?

+1

File: _tools/check_doc_imports.ts [hunk 1]
```diff
@@ -26,6 +26,11 @@ const RX_CODE_BLOCK = /`{3}([\w]*)\n([\S\s]+?)\n`{3}/gm;
 let shouldFail = false;
 let countChecked = 0;
 
+const allowedUrlPrefixes = [
+  "https://deno.land/std@$STD_VERSION/",
+  "https://html.spec.whatwg.org/",
```
Comment (lionel-rowe) [inline _tools/check_doc_imports.ts:31]: Is it OK to add a special case for this? Rationale — it's a JSON version of [a spec page provided by WHATWG](https://html.spec.whatwg.org/multipage/named-characters.html), which is the authoritative reference for named HTML entities and which states:

> _This list is static and [will not be expanded or changed in the future](https://github.com/whatwg/html/blob/main/FAQ.md#html-should-add-more-named-character-references)._

The alternative to importing directly from the WHATWG-provided JSON would be duplicating that JSON file within `std`.

**Comment by lionel-rowe** [comment]
Bikeshedding: encode/decode vs. escape/unescape vs [any other options]? I don't have a strong preference.

File: _tools/check_doc_imports.ts (hunk 1 above)
Comment (kt3k) [reply _tools/check_doc_imports.ts:31]: I'd prefer to duplicate the file in std with some note about the original. The spec doesn't seem guaranteeing the url and the shape of that JSON file.

**Comment by timreichen** [comment]
+1 for escape and unescape

File: _tools/check_doc_imports.ts (hunk 1 above)
Comment (lionel-rowe) [reply _tools/check_doc_imports.ts]: Fixed — one other advantage of importing internally is it doesn't have to use the same structure, which is more verbose than it needs to be for our purposes.

**Review by kt3k — APPROVED** [review]
LGTM

**Comment by kt3k** [comment]
Memo:

- php - [htmlspecialchars](https://www.php.net/manual/en/function.htmlspecialchars.php), [htmlspecialchars_decode](https://www.php.net/manual/en/function.htmlspecialchars-decode.php) (global builtins)
- ruby - [CGI.escapeHTML](https://docs.ruby-lang.org/en/3.2/CGI/Util.html#method-i-escapeHTML), [CGI.unescapeHTML](https://docs.ruby-lang.org/en/3.2/CGI/Util.html#method-i-unescape_html)
- golang - [EscapeString](https://pkg.go.dev/html#EscapeString), [UnescapeString](https://pkg.go.dev/html#UnescapeString) (html standard library)
- python - [html.escape](https://docs.python.org/3/library/html.html#html.escape), [html.unescape](https://docs.python.org/3/library/html.html#html.unescape) (html standard library)

**Comment by kt3k** [comment]
@lionel-rowe Thanks for your contribution!

**Pull request merged**